planet/src/main/webapp/places.js

```diff
@@ -315,9 +342,32 @@ function savePlace(x) {
     if(x.innerHTML === "favorite_border") {
         // Set as saved
         x.innerHTML = "favorite";
+        // Save place to database
+        let name = document.getElementById('place-name').innerHTML; 
```

Comment:
Wouldn't every card have the same id 'place-name', 'place-address', or 'openingTime'? How do we make sure we're getting the right information here?

Reply:
The onclick function is passing through "this" as a parameter meaning the current card!

Reply:
Ah I see. x is the current card button being clicked right? x is not used in  `document.getElementById()` (line346) though. Ignore this comment if you already tested this with multiple cards and it worked.

Reply:
+1 I share the concern about how this behaves with multiple cards.

In addition, it looks like we're misusing the html id property here. We probably want to use a class instead of id:

https://developer.mozilla.org/en-US/docs/Web/API/Element/id

"If the id value is not the empty string, it must be unique in a document."

Reply:
^^ I've tested the document.getElementByID() with multiple cards and it does return the correct value. I do understand the concern that all the id's are the same. I've tried switching it to use class-name but when trying to access document.getElementsByClass.innerHTML -> it returns undefined cause it seems that you can't only access the current element with the class name

Reply:
when getting elements by classname, you get an array back. If there's one element in the list that you want to access, you can get it by indexing the array: parentDOM.getElementsByClassName("test")[0];